source/MaterialXRuntime/RtNodeDef.cpp

```diff
@@ -214,6 +214,16 @@ const RtIdentifier& RtNodeDef::getNamespace() const
     return attr ? attr->asIdentifier() : EMPTY_IDENTIFIER;
 }
 
+void RtNodeDef::setInfo(const RtIdentifier& info) {
+    RtTypedValue* attr = prim()->createAttribute(Identifiers::INFO, RtType::STRING);
+    attr->asIdentifier() = info;
```

Comment:
The attribute is created as a string type (RtType::STRING). but then set as a RtIdentifier type (attr->asIdentifier()). Unfortunately this will not work and might result in hard to find bugs since the underlying storage is allocated for a string but assigned as an RtIdentifier.

If this is a documentation string it would be better to use a string both for the argument and for storage (here also changing the naming from "Info" to "Doc" as Bernard wrote):

```
void RtNodeDef::setDoc(const string& doc)
{
    RtTypedValue* attr = prim()->createAttribute(Identifiers::DOC, RtType::STRING);
    attr->asString() = doc;
}
```

Reply:
After discussing offline with @lyuborusev, it's should be clearer now: that the storage is as a string type (`RtType::STRING`) and the interface reflects this to be consistent.

Reply:
Excellent, thanks!